src/Microsoft.AspNetCore.Blazor/Json/SimpleJson/SimpleJson.cs

```diff
@@ -1434,10 +1465,22 @@ public virtual object DeserializeObject(object value, Type type)
                                 ?? throw new InvalidOperationException($"Cannot deserialize JSON into type '{type.FullName}' because it does not have a public parameterless constructor.");
                             obj = constructorDelegate();
 
+                            IDictionary<string, string> pascalCaseKeys = null;
+                            if (PropertyNaming == PropertyNamingSceme.CAMEL_CASE_PROPERTY_NAMING)
+                            {
+                                pascalCaseKeys = GetPascalCasePropertyNames(jsonObject.Keys);
+                            }
+
                             foreach (KeyValuePair<string, KeyValuePair<Type, ReflectionUtils.SetDelegate>> setter in SetCache[type])
                             {
+                                string setterKey = null;
+                                if (PropertyNaming == PropertyNamingSceme.CAMEL_CASE_PROPERTY_NAMING)
```

Comment:
@SteveSandersonMS I'm abit fuzzy on how the deserialize part should be done to be case sensitive again. I need abit of clarification on that part.

So far what I did is transferred the computation to the last part where the value class properties is set to the deserialized json value is done instead of blindly converting the property names to pascal case like last time. This one makes a mapping to the ```setter.key => JSON.key``` instead and gets the keys there only when the property naming is ```CamelCase```.

The mapping could have been done with ```Linq.Select``` but I thought otherwise as to follow how SimpleJson's structure was done.